docs/docs/pricing/README.md

```diff
@@ -9,137 +9,150 @@ We believe anyone should be able to run simple, low-volume workflows at no cost.
 
-Pipedream counts an **invocation** each time a workflow or event source is triggered by an incoming event.
+:::
 
-Pipedream increments the count of invocations by one for each incoming event, regardless of the number of steps in your workflow. For example, if you send an HTTP request to a workflow with five steps, Pipedream will count that as one invocation.
+::: details A workflow with 512 megabytes of memory running for 5 seconds
 
-Moreover, if you have a workflow triggered by a cron job running once a minute for the entire day, that will incur 1,440 invocations (60 minute \* 24 hours). For cron-triggered workflows or event sources, remember that you can always reduce the frequency to reduce your invocations.
+2 credits are incurred, because the workflow executed for a total of 5 seconds at 512 megabytes of memory.
+:::
 
-If an event emitted by an event source triggers a single workflow, that will count as **two** invocations: one for the source, and one for the workflow. In other words, source and workflow execution is distinct: each counts invocations on its own.
 
-Your workflow's [memory settings](/workflows/settings/#memory) also impact the number of invocations you're charged for each workflow execution. [Read more here](#how-does-workflow-memory-affect-billable-invocations).
+#### Source Credit Usage
 
-#### Scenarios
+The first credit used during a [source's](/docs/sources/) execution is included for free.
+
+This means that the first {{ $site.themeConfig.base_credits_price.seconds }} of computation on an source do not incur credits. This behavior includes [Free Tier](/pricing/#free-tier) accounts.
+
+::: tip
 
-::: details Webhook triggered workflow
+This free first credit per execution **only** applies to sources from the [Pipedream public registry](/docs/sources). 
 
-*1* invocation is incurred per HTTP webhook. The HTTP endpoint is *not* considered a source.
+If you deploy a private custom source to your account, then all computation time including the inital {{ $site.themeConfig.base_credits_price.seconds }} seconds for that private source counted toward credits.
 
 :::
 
-::: details Scheduled workflow
+::: details A polling source finishing under {{ $site.themeConfig.base_credits_price.seconds }} seconds per execution
 
-*1* invocation is incurred per new event emitted by the source.
-*1* invocation is incurred per new event processed by the workflow.
+For example, a source that polls an API for new events like [Airtable - New Row Added](https://pipedream.com/apps/airtable/triggers/new-records) only takes ~5 seconds to poll and emit events to subscribing workflows.
 
-Example: a schedule triggered workflow is configured to run every 15 minutes. Two invocations are incurred every 15 minutes. One from the timer source emitting an event, and the other from the workflow processing the event.
+This would result 0 credits. Regardless of the number of times the source polled because:
```

Comment:
"This would result in 0 credits per run:", and I think you can remove "Regardless of the number of times the source polled because:"